app/src/main/java/com/junkfood/seal/util/FileUtil.kt

```diff
@@ -108,15 +110,48 @@ object FileUtil {
         val path: String = treeUri.path.toString()
         Log.d(TAG, path)
         if (!path.contains("primary:")) {
-            TextUtil.makeToast("This directory is not supported")
-            return File(
-                Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).absolutePath,
-                "Seal"
-            ).absolutePath
+            //Get the SD ID that is after two "/" and before an ":"
+            val sdCardId = path.substring(path.indexOf("/", 1) + 1, path.indexOf(":"))
+            Log.d(TAG, "SD Card ID: $sdCardId")
+            //Get the path after the SD ID
+            val pathAfterSdId =
+                path.substring(path.indexOf(sdCardId) + sdCardId.length + 1, path.length)
+
+            val pathExists: Boolean = File("/storage/$sdCardId/$pathAfterSdId").exists()
+            Log.d(TAG, "Path exists: $pathExists")
+
+            return "/storage/$sdCardId/$pathAfterSdId"
         }
+
         val last: String = path.split("primary:").last()
         return "/storage/emulated/0/$last"
     }
 
+    //create a file in the indicated directory and write some data in it
+    fun createFileInDirectory(directoryUri: Uri, fileName: String, data: String) {
```

Comment:
What's the `data` here?

Reply:
This were just a test function. Can be deleted.